File: project-euler.java/src/main/java/euler/solutions/_7.java
package euler.solutions;

import euler.common.BaseSolution;

import static euler.common.NumberUtils.PrimesIterator;

public class _7 extends BaseSolution {

    @Override
    public long solve() {
        PrimesIterator primes = new PrimesIterator();
        for (int i = 4; i < 10001; i++)
            primes.next();

        return primes.next();
    }
}

File: project-euler.go/solutions/_7/7.go
package _7

import "project-euler/common/numbers"

func Solve() int64 {
	primes := numbers.NewPrimesGenerator()

	n := primes.Next()
	for i := 4; i < 10001; i++ {
		n = primes.Next()
	}

	return int64(n)
}
